### PR TITLE
fix: chat DM not allow change image/name info PING-4086

### PR DESCRIPTION
**Pull request by GarryLaly**
# Helio PR Checklist

## Please review all of the following checks before making PR

- [x] Jika menambahkan sebuah komponen, apakah sudah dipastikan tidak ada komponen yang redundant, baik secara UI maupun fungsional?
- [x] Periksa apakah komponen yang dibuat reusable (contoh: membuat komponen View untuk membuat Divider antar section)
- [x] Periksa untuk penggunaan state management yang baik dan benar (contoh: Context, useState), terutama penggunaan local state untuk optimistic UI (Pengoperan state dan data melalui props dari parent ke children dan sebaliknya)
- [x] Pastikan komponen yang dibuat sudah sesuai dengan module yang ada terutama penamaan dan penempatan untuk meningkatkan readability dan maintainability.
- [x] Implementasi fallback error untuk menangani kesalahan tak terduga, mencegah aplikasi mengalami crash
- [x] Evaluasi kompleksitas kode dan perbaiki logika yang kompleks dengan memecahnya menjadi fungsi atau komponen yang lebih kecil dan lebih mudah dimengerti.
- [x] Apakah sudah resolve semua komentar dari CodeRabbit🐇?
- [x] Apakah semua unit tests sudah Pass/Success? (Tidak ada bypass)
- [x] Periksa semua perubahan teks Bahasa Inggris approved oleh Bastian?
- [x] Apakah sudah menggunakan normalized.dimen/ normalizeFontSize untuk semua perubahan style?
- [x] Pastikan untuk mengikuti [guidelines pembuatan komponen Guidelines About Loading and Layout](https://docs.google.com/document/d/1GHbvEtFrQmEQXYBV9dLrPI654n2DbPwS1qjedajLSiw/edit)

### Author's comment

if you have any unchecked items, please explain the reason why.

### Summary by CodeRabbit

- Refactor: Enhanced the `ChatInfoScreen` component to handle image clicks and name changes based on the channel type, improving user interaction.
- New Feature: Added conditional rendering of the pencil icon for name changes depending on the channel type, enhancing the user interface.

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 9b059595b04e36f884f65b106d59d56b304943da and a758b70a70b658be8b31b23ef8918fa793a95a7d commits.
</details>

<details>
<summary>Files selected (1)</summary>

* src/screens/ChatInfoScreen/ChatInfoScreen.tsx (2)
</details>






---

File: src/screens/ChatInfoScreen/ChatInfoScreen.tsx [hunk 1]
```diff
@@ -310,7 +310,8 @@ const ChatInfoScreen = () => {
         <View style={styles.lineTop} />
         <TouchableOpacity
           testID="imageClick"
-          onPress={handleOnImageClicked}
+          activeOpacity={channelInfo?.channelType === CHANNEL_GROUP ? 0.2 : 1}
+          onPress={channelInfo?.channelType === CHANNEL_GROUP ? () => handleOnImageClicked() : null}
```
Comment (bastian-purrer) [inline src/screens/ChatInfoScreen/ChatInfoScreen.tsx:314]: Image description   CodeRabbit

The changes made here are good for user experience. It prevents users from clicking on the image in a group chat, which is a sensible restriction. However, it's better to move the condition check inside the `handleOnImageClicked` function. This way, you can encapsulate the logic within the function itself and keep the JSX cleaner.
```diff
- activeOpacity={channelInfo?.channelType === CHANNEL_GROUP ? 0.2 : 1}
- onPress={channelInfo?.channelType === CHANNEL_GROUP ? () => handleOnImageClicked() : null}
+ activeOpacity={1}
+ onPress={() => handleOnImageClicked()}
```
And then in your `handleOnImageClicked` function:
```diff
function handleOnImageClicked() {
+   if (channelInfo?.channelType !== CHANNEL_GROUP) {
+     return;
+   }
  // existing code...
}
```

File: src/screens/ChatInfoScreen/ChatInfoScreen.tsx [hunk 2]
```diff
@@ -329,13 +330,15 @@ const ChatInfoScreen = () => {
...
-          </TouchableOpacity>
+          {channelInfo?.channelType === CHANNEL_GROUP && (
+            <TouchableOpacity onPress={handleOpenNameChange} style={styles.pencilIconTouchable}>
+              {isUpdatingName ? (
+                <ActivityIndicator />
+              ) : (
+                <IconPencil width={20} height={20} color={COLORS.gray400} />
+              )}
+            </TouchableOpacity>
+          )}
```
Comment (bastian-purrer) [inline src/screens/ChatInfoScreen/ChatInfoScreen.tsx:341]: Image description   CodeRabbit

Similar to the previous comment, the conditional rendering of the pencil icon based on the channel type is a good UX decision. However, I would suggest moving this logic into the `handleOpenNameChange` function to keep the JSX cleaner.
```diff
- {channelInfo?.channelType === CHANNEL_GROUP && (
-   <TouchableOpacity onPress={handleOpenNameChange} style={styles.pencilIconTouchable}>
-     {isUpdatingName ? (
-       <ActivityIndicator />
-     ) : (
-       <IconPencil width={20} height={20} color={COLORS.gray400} />
-     )}
-   </TouchableOpacity>
- )}
+ <TouchableOpacity onPress={handleOpenNameChange} style={styles.pencilIconTouchable}>
+   {isUpdatingName ? (
+     <ActivityIndicator />
+   ) : (
+     <IconPencil width={20} height={20} color={COLORS.gray400} />
+   )}
+ </TouchableOpacity>
```
And then in your `handleOpenNameChange` function:
```diff
function handleOpenNameChange() {
+   if (channelInfo?.channelType !== CHANNEL_GROUP) {
+     return;
+   }
  // existing code...
}
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 9b059595b04e36f884f65b106d59d56b304943da and a758b70a70b658be8b31b23ef8918fa793a95a7d commits.
</details>

<details>
<summary>Files selected (1)</summary>

* src/screens/ChatInfoScreen/ChatInfoScreen.tsx (2)
</details>








<details>
<summary>Review comments generated (2)</summary>

* Review: 2
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by fajarism — APPROVED** [review]

**Pull request merged**